src/category/category.service.ts

```diff
@@ -0,0 +1,63 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { FindOptionsWhere } from 'typeorm';
+
+import { categoryMessages } from '@common/constants/messages';
+import { CreateCategoryDto } from './dto/create-category.dto';
+import { UpdateCategoryDto } from './dto/update-category.dto';
+import { CategoryRepository } from './repositories/category.repository';
+import { Category } from './entities/category.entity';
+
+@Injectable()
+export class CategoryService {
+  constructor(private readonly categoryRepository: CategoryRepository) {}
+
+  async findAll() {
+    const categories = await this.categoryRepository.find();
```

Comment:
If you dont want to use try catch here, you can just do a `return this.categoryRepository.find();` and remove the async from the method signature

Reply:
done